utils/cluster_shrinker/cluster_shrinker.pony

```diff
@@ -0,0 +1,136 @@
+/*
+
+Copyright 2017 The Wallaroo Authors.
+
+Licensed as a Wallaroo Enterprise file under the Wallaroo Community
+License (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+     https://github.com/wallaroolabs/wallaroo/blob/master/LICENSE
+
+*/
+
+"""
+A tool for sending autoscale shrink messages.
+"""
+use "buffered"
+use "net"
+use "files"
+use "wallaroo_labs/bytes"
+use "wallaroo_labs/messages"
+use "wallaroo_labs/options"
+
+actor Main
+  var _conn: (TCPConnection | None) = None
+
+  new create(env: Env) =>
+    try
+      var x_host: String = ""
+      var x_service: String = "0"
+      var message: String = ""
+      let options = Options(env.args)
+
+      options
+        .add("external", "e", StringArgument)
+        .add("message", "m", StringArgument)
+        .add("help", "h", None)
+
+        for option in options do
+          match option
+          | ("external", let arg: String) =>
+            let x_addr = arg.split(":")
+            x_host = x_addr(0)?
+            x_service = x_addr(1)?
+          | ("message", let arg: String) => message = arg
+          | ("help", None) =>
+            @printf[I32](
+              """
+              PARAMETERS:
+              -----------------------------------------------------------------------------------
+              --external/-e [Specifies address to send message to]
+              --message/-m [Specifies message contents to send]
+                  Specify names of nodes or number of nodes.
+                  If 1st char is a digit, specify number of of nodes;
+                  else specify comma-separated list of node names.
+              -----------------------------------------------------------------------------------
+              """.cstring())
+            return
+          end
+        end
+
+      let auth = env.root as AmbientAuth
+      (let query: Bool, let node_names: Array[String], let num_nodes: USize) =
+        parse_shrink_cmd_line(message)?
+      let msg = ExternalMsgEncoder.shrink(query, node_names, num_nodes)
+      let tcp_auth = TCPConnectAuth(auth)
+      _conn = TCPConnection(tcp_auth, ExternalSenderConnectNotifier(auth,
+        msg, query), x_host, x_service)
+    else
+      @printf[I32]("Error sending.\n".cstring())
+    end
+
+    fun parse_shrink_cmd_line(s: String): (Bool, Array[String], USize) ? =>
```

Comment:
this seems to process the `message` not the entire command line. I think the name of this fun is misleading.